toolkit/tools/pkg/imagecustomizerlib/configvalidation.go

```diff
@@ -503,15 +524,75 @@ func validateIsoPxeCustomization(rc *ResolvedConfig) error {
 		// While defining a storage configuration can work when the input image is
 		// an iso, there is no obvious point of moving content between partitions
 		// where all partitions get collapsed into the squashfs at the end.
-		if rc.Config.CustomizePartitions() {
+		if rc.Storage.CustomizePartitions() {
 			return ErrCannotCustomizePartitionsOnIso
 		}
 	}
 
 	return nil
 }
 
-func resolveOutputArtifacts(configChain []*ConfigWithBasePath) *imagecustomizerapi.Artifacts {
+func validateStorage(configChain []*ConfigWithPath) error {
+	var baseHasDisks bool
+
+	for _, configWithBase := range configChain {
+		storage := configWithBase.Config.Storage
+
+		hasDisks := storage.CustomizePartitions()
+		hasResetUUID := storage.ResetPartitionsUuidsType != imagecustomizerapi.ResetPartitionsUuidsTypeDefault
```

Comment:
nit: these local variables are redundant to have, can change it to something like:

```
if storage.ResetPartitionsUuidsType != imagecustomizerapi.ResetPartitionsUuidsTypeDefault {
```